Source/NSCollectionView.m

```diff
@@ -26,37 +29,81 @@ Frank LeGrand (frank.legrand@testplant.com)
 */
 
 #import "Foundation/NSKeyedArchiver.h"
+
 #import <Foundation/NSGeometry.h>
 #import <Foundation/NSIndexSet.h>
+#import <Foundation/NSSet.h>
+#import <Foundation/NSDictionary.h>
+#import <Foundation/NSMapTable.h>
 #import <Foundation/NSKeyedArchiver.h>
 
 #import "AppKit/NSApplication.h"
 #import "AppKit/NSClipView.h"
 #import "AppKit/NSCollectionView.h"
 #import "AppKit/NSCollectionViewItem.h"
+#import "AppKit/NSCollectionViewLayout.h"
+#import "AppKit/NSCollectionViewGridLayout.h"
 #import "AppKit/NSEvent.h"
 #import "AppKit/NSGraphics.h"
 #import "AppKit/NSImage.h"
 #import "AppKit/NSKeyValueBinding.h"
+#import "AppKit/NSNib.h"
 #import "AppKit/NSPasteboard.h"
 #import "AppKit/NSWindow.h"
+
 #import "GSGuiPrivate.h"
+#import "GSFastEnumeration.h"
 
 #include <math.h>
 
 static NSString* NSCollectionViewMinItemSizeKey              = @"NSMinGridSize";
 static NSString* NSCollectionViewMaxItemSizeKey              = @"NSMaxGridSize";
-//static NSString* NSCollectionViewVerticalMarginKey           = @"NSCollectionViewVerticalMarginKey";
 static NSString* NSCollectionViewMaxNumberOfRowsKey          = @"NSMaxNumberOfGridRows";
 static NSString* NSCollectionViewMaxNumberOfColumnsKey       = @"NSMaxNumberOfGridColumns";
 static NSString* NSCollectionViewSelectableKey               = @"NSSelectable";
 static NSString* NSCollectionViewAllowsMultipleSelectionKey  = @"NSAllowsMultipleSelection";
 static NSString* NSCollectionViewBackgroundColorsKey         = @"NSBackgroundColors";
+static NSString* NSCollectionViewLayoutKey                   = @"NSCollectionViewLayout";
+
+static NSCollectionViewSupplementaryElementKind GSNoSupplementaryElement  = @"GSNoSupplementaryElement"; // private
 
 /*
  * Class variables
  */
-static NSString *placeholderItem = nil;
+static NSString *_placeholderItem = nil;
+
+@interface NSObject (CollectionViewInternalPrivate)
+
+/**
+ * This method is used to determine if the receiver overrides the
+ * given selector.
+ */
+- (BOOL) overridesSelector: (SEL)s;
+
+@end
+
+@implementation NSObject (CollectionViewInternalPrivate)
+
+- (BOOL) overridesSelector: (SEL)s
```

Comment:
Are you sure you need the full complexity of this method here? It seems to get used only once. You could have a look at the way I solved a similar issue in NSDocument.m line 88.

Reply:
I am wondering if it should not be moved to GNUstepBase/NSObject+GNUstepBase.*.   Since it is generally useful and, since you solved a similar issue (I haven't looked at NSDocument yet...) would it be better to move it to someplace central?

Reply:
Added analogous macro OVERRIDDEN per suggestion. Resolving.